### ISSUE TITLE
fixing regression in code

### ISSUE THREAD

**Issue by jeromeetienne** [jeromeetienne/chess_ai.ml#2]
- garanteed good result - https://github.com/jeromeetienne/chess_ai.ml/commit/21c320964d730e8893443d2df319565867e8a9dc
- good commit
   - https://github.com/jeromeetienne/chess_ai.ml/commit/523bf316f594192ff37978191c7759cced7ecf03
   - https://github.com/jeromeetienne/chess_ai.ml/commit/3cdff1c7b2c60a9b85f9d9999c111dff09d5b264
   - https://github.com/jeromeetienne/chess_ai.ml/commit/523bf316f594192ff37978191c7759cced7ecf03
   - https://github.com/jeromeetienne/chess_ai.ml/commit/ac193862e8de9a9bb11033b2d4fa858bc9e85721 Oct 18, 2025
   - https://github.com/jeromeetienne/chess_ai.ml/commit/b257357d4592b32900db4a7ed1509283546b4ab9
   - https://github.com/jeromeetienne/chess_ai.ml/commit/5f71cf572ceca5c00ff0eba6007fa8dba1b975aa
   - https://github.com/jeromeetienne/chess_ai.ml/commit/35c04889968017099e525e587ec15dc35d0aa6df
   - https://github.com/jeromeetienne/chess_ai.ml/commit/3c402f8c300ca91795f48f62b1cecb12548f350c Oct 19, 2025

## Process
- test at the end of each test since `Oct 10, 2025` and then disect during the same day
- if a commit got the bug, double check the modification. do the modification twice to be sure

## `bin/train.py` 
- enable 42 random seed
  - `torch.manual_seed(42)`
- restore the lr scheduler and early-stop https://github.com/jeromeetienne/chess_ai.ml/blob/21c320964d730e8893443d2df319565867e8a9dc/src/train.py#L131-L133
- restore old loss weighting https://github.com/jeromeetienne/chess_ai.ml/blob/efe0f1abd1bd657b5821a893e4e1c2486b103ae1/bin/train.py#L244-L251
- use `SmoothL1Loss`

```python
criterion_reg = torch.nn.SmoothL1Loss()
```

## src/encoding/move_encoding.py
- enable uci2class move encoding, remove alpha zero

## src/libs/chess_model.py
- restore default model architecture - ChessModelConv2d https://github.com/jeromeetienne/chess_ai.ml/blob/3cdff1c7b2c60a9b85f9d9999c111dff09d5b264/src/libs/chess_model.py#L44-L50

```python
params = ChessModelConv2dParams(
    conv_out_channels=[64, 128, 256],
    cls_fc_size=128,
    reg_fc_size=64,
    cls_head_dropout=0.5,
    reg_head_dropout=0.5,
)
```

## Expected results
- cmdline to run `./bin/train.py -fc 20 -lr 0.0005`
- going deep at 10 epochs - accuracy a little higher that epoch numbers

<img width="1558" height="589" alt="Image" src="https://github.com/user-attachments/assets/35d312a8-74e0-4387-8ac0-2d40075f6a68" />

<img width="1600" height="400" alt="Image" src="https://github.com/user-attachments/assets/10d030b1-db88-4bd6-8f05-b51aa42b4b75" />

<img width="680" height="560" alt="Image" src="https://github.com/user-attachments/assets/628f2de3-2ca0-43ca-ab53-616133905b9a" />
